source/common/http/codec_helper.h

```diff
@@ -40,7 +48,14 @@ class StreamCallbackHelper {
     callbacks_.reserve(8);
   }
 
-  void addCallbacks_(StreamCallbacks& callbacks) { callbacks_.push_back(&callbacks); }
+  void addCallbacks_(StreamCallbacks& callbacks) {
+    callbacks_.push_back(&callbacks);
+    if (!reset_callbacks_started_) {
```

Comment:
Does this actually happen? I feel like adding callbacks after reset is broken if it's happening. If this is just to have symmetry with other checks but doesn't happen can we switch to ASSERT?